src/components/grid/Grid.tsx

```diff
@@ -33,16 +51,65 @@ const Grid: React.FC<PropsGrid> = ({ board, ws, yourTurn }) => {
                 }
               />
             );
+            gridNew[key] = [...(gridNew[key] || []), elementHitbox];
           } else if (square.isPlaced) {
-            gridNew.push(<Wall coordinates={coordinates} color="orange" />);
+            const elementWall = <Wall coordinates={coordinates} color="orange" />;
+            gridNew[key] = [...(gridNew[key] || []), elementWall];
+          }
+        } else if (square.type === SquareTypes.Player) {
+          if (square?.playerId === playerID) {
+            if (yourTurn) {
+              const elementPlayerTile = (
+                <PlayerTile
+                  coordinates={coordinates}
+                  handleOnClick={() => setIsPressed((prevIsPressed) => !prevIsPressed)}
+                />
+              );
+              gridNew[key] = [...(gridNew[key] || []), elementPlayerTile];
+            }
+            const elementGargoylePlayer = <GargoylePlayer isRotated={!isPlayerHost} coordinates={coordinates} />;
+            gridNew[key] = [...(gridNew[key] || []), elementGargoylePlayer];
+          } else if (square?.playerId && square?.playerId !== playerID) {
```

Comment:
I don't think you need to check `square?.playerId !== playerID`, because the previous if statement already did